CMake/CheckAndAddFlag.cmake

```diff
@@ -27,13 +27,23 @@ function(check_and_add_flag var flag)
     message(FATAL_ERROR "check_and_add_flag called with incorrect arguments: ${ARGN}")
   endif()
 
+  set(is_c "$<COMPILE_LANGUAGE:C>")
+  set(is_cxx "$<COMPILE_LANGUAGE:CXX>")
+
+  # The Visual Studio generators don't support COMPILE_LANGUAGE
+  # So we fail all the C flags and only actually test CXX ones
+  if(CMAKE_GENERATOR MATCHES "Visual Studio")
```

Comment:
I was also thinking, it might make sense to split this whole function up between `check_and_add_cxx_flag` and `check_and_add_flag` (which only checks against C but adds to both), since practically all supported C flags should also work for C++ (only case I can think of where this isn't true is something like `-std=c11`). An added benefit would be faster configuration time due to less calls out to the build system.